src/lib/decorators.d.ts

```diff
@@ -17,6 +17,9 @@ type DecoratorContext =
     | ClassMemberDecoratorContext
     ;
 
+type DecoratorMetadata =
+    typeof globalThis extends { Symbol: { readonly metadata: symbol } } ? object : object | undefined;
+

```

Comment:
Unfortunately, this ends up being hard to use in practice. We should probably change the definition somewhat:

```suggestion
type DecoratorMetadataObject = Record<PropertyKey, unknown> & object;
type DecoratorMetadata =
    typeof globalThis extends { Symbol: { readonly metadata: symbol } } ? DecoratorMetadataObject : DecoratorMetadataObject | undefined;

interface Function {
  [Symbol.metadata]: DecoratorMetadata | null;
}
```